backend/src/impl/default_controllers_impl.py

```diff
@@ -99,16 +105,16 @@ def datasets_get(
     page_size: int,
 ) -> DatasetsReturn:
     parsed_dataset_ids = dataset_ids.split(",") if dataset_ids else None
-    return DatasetMetaDataModel.find(
+    return DatasetDBUtils.find_datasets(
         page, page_size, parsed_dataset_ids, dataset_name, task
     )
 
 
 """ /systems """
 
 
-def systems_system_id_get(system_id: str) -> SystemModel:
-    system = SystemModel.find_one_by_id(system_id)
+def systems_system_id_get(system_id: str) -> System:
+    system = DBUtils.find_one_by_id(DBUtils.DEV_SYSTEM_METADATA, system_id)
```

Comment:
I think `system_from_dict()` needs to be applied to `system` here so it returns a `System` object.